salt-file-protocol-file-listing-details.md

```diff
@@ -0,0 +1,60 @@
+- Feature Name: salt-file-protocol-file-listing-details
+- Start Date: 20200723
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+cp.list_master does not provide enough information to determine if files on the minion are up to data, and if they should still exits. This proposal is to look into expanding the information available on files and directories. This would lead to cp.get_dir performance improvements.
+
+# Motivation
+[motivation]: #motivation
+
+salt:// file fetching does not support syncing files between the master and minion in an efficient  manner. For example deleting the files in the cache and re-fetching the files is often used to ensure what is on the minion matches what is on the master.
+
+By providing more detail information about files and folders it should be possible to reduce the time it takes to sync files between the master and the minion. Agree to data structure to return is just the first step. This SEP is about agreeing to the data that is required.
+
+# Design
+[design]: #detailed-design
+The following is the proposal
+```
+cp.list_master detail=True directories_only=False path=/
+'.':
+   Time: seconds since 1970
```

Comment:
I assume this is supposed to be last modification time? Not creation or last access.
I also assume this is Unix epoch time, in which case the precise specification is since 1970-01-01T00:00:00.000+00 excluding leap seconds.